polus-precompute-volume-plugin/src/main.py

```diff
@@ -0,0 +1,86 @@
+import logging, argparse, time, multiprocessing, traceback
+
+from concurrent.futures import ThreadPoolExecutor, ProcessPoolExecutor
+from itertools import repeat
+
+import utils
+import os
+from pathlib import Path
+
+import filepattern
+from filepattern import FilePattern as fp
+
+import numpy as np
+
+# Initialize the logger    
+logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+                    datefmt='%d-%b-%y %H:%M:%S')
+logger = logging.getLogger("main")
+logger.setLevel(logging.INFO)
+
+def main(input_dir: str,
+         output_dir: str,
+         imagetype: str,
+         imagepattern: str,
+         mesh: bool):
+    
+    # Get list of images that we are going to through
+    # Get list of output paths for every image
+    logger.info("\n Getting the images...")
+    fp_images = fp(Path(input_dir),imagepattern)
+    input_images = []
+    output_images = []
+    for i in fp_images():
+        image = i[0]['file']
+        if os.path.exists(image):
+            input_images.append(image)
+            output_images.append(os.path.join(output_dir, os.path.basename(image)))
+    assert len(input_images) == len(output_images)
+
+    # Build one pyramid for each image in the input directory
+    with ProcessPoolExecutor(max_workers=2) as executor:
+        executor.map(utils.build_pyramid, input_images, output_images, repeat(imagetype), repeat(mesh))
```

Comment:
I vaguely remember having this conversation about using 2 workers, but I don't remember why we landed on that. Can you provide a brief comment about why 2 was selected? Why shouldn't we scale this with the number of cores on the machine?